platforms/documentation/docs/src/docs/userguide/dep-man/declaring-depedencies/dependency_versions.adoc

```diff
@@ -0,0 +1,372 @@
+* **Comparing Equivalent Parts:**
+** **Numeric vs. Numeric:** Higher numeric value is considered higher: `1.1 < 1.2`.
+** **Numeric vs. Non-numeric:** Numeric parts are higher than non-numeric parts: `1.a < 1.1`.
+** **Non-numeric vs. Non-numeric:** Parts are compared alphabetically and case-sensitively: `1.A < 1.B < 1.a < 1.b`.
+** **Extra Numeric Part:** A version with an additional numeric part is higher, even if it's zero: `1.1 < 1.1.0`.
+** **Extra Non-numeric Part:** A version with an extra non-numeric part is lower: `1.1.a < 1.1`.
+* **Special Non-numeric Parts:**
+** `dev` is *lower* than any other non-numeric part: `1.0-dev < 1.0-ALPHA < 1.0-alpha < 1.0-rc`.
+** `rc`, `snapshot`, `final`, `ga`, `release`, and `sp` are *higher* than any other string part, in this order: `1.0-zeta < 1.0-rc < 1.0-snapshot < 1.0-final < 1.0-ga < 1.0-release < 1.0-sp`.
+** These special values are **not case-sensitive** and their ordering does not depend on the separator used: `1.0-RC-1` == `1.0.rc.1`.
+
+[[rich-version-constraints]]
+== Declaring rich versions
+
+When you declare a version using the shorthand notation, then the version is considered a <<#sec:required-version,required version>>:
+
+====
+include::sample[dir="snippets/dependencyManagement/declaringDependencies-concreteVersion/kotlin",files="build.gradle.kts[tags=required-version]"]
+include::sample[dir="snippets/dependencyManagement/declaringDependencies-concreteVersion/groovy",files="build.gradle[tags=required-version]"]
+====
+
+This means the minimum version will be `1.7.15` and it can be optimistically upgraded by the engine.
+
+To enforce a <<sec:strict-version,strict version>> and ensure that only the specified version of a dependency is used, rejecting any other versions even if they would normally be compatible:
+
+====
+include::sample[dir="snippets/dependencyManagement/declaringDependencies-concreteVersion/kotlin",files="build.gradle.kts[tags=rich-version-intro]"]
+include::sample[dir="snippets/dependencyManagement/declaringDependencies-concreteVersion/groovy",files="build.gradle[tags=rich-version-intro]"]
+====
+
+Gradle supports a model for *rich version* declarations, allowing you to combine different levels of version specificity.
+
+The key terms, listed from strongest to weakest, are:
+
+[[sec:strict-version]]
+`strictly` or `!!`::
+This is the strongest version declaration.
+Any version not matching this notation will be excluded.
+If used on a declared dependency, `strictly` can downgrade a version.
+For transitive dependencies, if no acceptable version is found, dependency resolution will fail.
++
+Dynamic versions are supported.
++
+When defined, it overrides any previous `require` declaration and clears any previous `reject` already declared on that dependency.
+
+[[sec:required-version]]
+`require`::
```

Comment:
💭 Not a comment on the docs, but I personally find this name somewhat confusing.  "Require" sounds very similar to "strict" in common usage.  I wish this was called "atleast" or "minimal" or something.